Bump Gradle wrapper to 8.8

wzieba opened the pull request:
<!-- Remember about a good descriptive title. -->
### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR bumps Gradle wrapper to 8.8.

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

CI checks should be just fine.


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

wzieba committed:
Bump Gradle wrapper to 8.8

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,7 @@
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2.1-all.zip
+distributionSha256Sum=a4b4158601f8636cdeeab09bd76afb640030bb5b144aafe261a5e8af027dc612
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-bin.zip
```
Question (❓): I noticed this changing from `all` (complete) to `bin` (binary-only) distribution type. Was that done on purpose, just because the `bin` is smaller in size and quicker to download maybe, is that really that noticable? 🤔

PS: I am asking that because I am usually in favor of the `all` distribution type, it containing the full documentation, source code, and other resources.

wzieba replied:
It was intentional, but I'm not insisting on using `bin` - we can continue using `all`. I've addressed it in c6d995e

ParaskP7 replied:
👋  Wojtek and thanks!

PS: If that was intentional and for good reason, I just wanted you to share that and we could have gone the `bin` way, I'm not insisting on using `all` as well, I was just mentioning why I am in favor of it. Just let me know and we can totally flip that. I'll resolve this for now.

ParaskP7 reviewed (approved):
👋 @wzieba !

I have reviewed this PR, everything works as expected, good job! 🌟 

I also tested this with included builds (`local-builds.gradle`) and everything seems to be working as expected. ✅ 

Btw, why are we doing this update now, I am just trying to understand whether there was a need for that, for security purposes maybe, or whether this was done purely to keep our tooling up-to-date, and maybe, to also have Gradle prepared for a future AGP update? 🤔

-----

I have left a question (❓) for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

-----

`EXTRAS`

1. I issued another such update command and it gave me an updated `gradlew` and `gradlew-wrapper.jar` file. Can you try it again on your side and see if you also are getting these file to be re-updated, just like it happens on my side? 🙏 

FYI: I used the following command:
`./gradlew wrapper --gradle-version=8.8 --distribution-type=bin --gradle-distribution-sha256-sum=a4b4158601f8636cdeeab09bd76afb640030bb5b144aafe261a5e8af027dc612`

wzieba committed:
Use `all` Gradle distribution

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/224">19.4</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba commented:
> Btw, why are we doing this update now, I am just trying to understand whether there was a need for that, for security purposes maybe, or whether this was done purely to keep our tooling up-to-date, and maybe, to also have Gradle prepared for a future AGP update? 🤔

It's a preparation for AGP update, but also a casual dependency update. I think that we were usually combining AGP and Gradle updates, but there's no need to not update Gradle separately and more often.

> I issued another such update command and it gave me an updated gradlew and gradlew-wrapper.jar file. Can you try it again on your side and see if you also are getting these file to be re-updated, just like it happens on my side? 🙏

I've doubled check and there's no difference for my `gradlew` and `gradlew-wrapper.jar` files. Also, I think "Validate Gradle Wrapper" should throw an error if these files were not precisely as they intend to be 🤔 . Could you please double check if there's still a diff when you run `wrapper` task?

ParaskP7 commented:
> It's a preparation for AGP update, but also a casual dependency update. I think that we were usually combining AGP and Gradle updates, but there's no need to not update Gradle separately and more often.

I agree, maybe in the past, with some major updates, we had to do both, but unless we do have to do both, let's always do Gradle in isolation, and only then follow-up on AGP updates. 💯 

> I've doubled check and there's no difference for my gradlew and gradlew-wrapper.jar files. Also, I think "Validate Gradle Wrapper" should throw an error if these files were not precisely as they intend to be 🤔 . Could you please double check if there's still a diff when you run wrapper task?

Yea, I did double and triple check that, me running `./gradlew wrapper --gradle-version=8.8 ...` will always produce a new diff. On the `gradlew` side it is just a comment change (can be ignored I guess), but on the `gradlew-wrapper.jar` side I am also seeing some code changes (can't be too much ignored). 🤔 

 I wonder why I get a diff and you don't, this is so strange and unpredictable, and imagine, we both use a Mac... 🤷 

-----

FYI: I don't want to block the merge so feel free to merge this anyway. 😊 

PS: Same for [WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/pull/21015).

wzieba commented:
> Yea, I did double and triple check that, me running ./gradlew wrapper --gradle-version=8.8 ... will always produce a new diff.

This is concerning 😕 - could you please prepare a PR (or only branch) with the diff? I'd like to examine the difference.

ParaskP7 committed:
Build: Upgrade gradle to 8.8

Command: ./gradlew wrapper --gradle-version=8.8 --distribution-type=all
--gradle-distribution-sha256-sum=
f8b4f4772d302c8ff580bc40d0f56e715de69b163546944f787c87abf209c961

ParaskP7 commented:
Sure thing @wzieba I just did that and pushed this commit [here](https://github.com/woocommerce/woocommerce-android/commit/3e70c000cbfc90722bb3000c521f176bca216095) on this [bump_gradle_wrapper_petros](https://github.com/woocommerce/woocommerce-android/commits/bump_gradle_wrapper_petros/) branch. 👍 

Let me know if that helps. 🙏

wzieba commented:
So both versions are safe:

From this PR: 
<img width="796" alt="image" src="https://github.com/woocommerce/woocommerce-android/assets/5845095/0be0d0f6-6192-4cd7-a5df-ee02eda68b1b">

From your commit
<img width="801" alt="image" src="https://github.com/woocommerce/woocommerce-android/assets/5845095/6004bc8b-313a-48f0-9f25-77dbd32e7eb7">

But comparing this to https://gradle.org/release-checksums/, it looks like my update updated jar to version 8.6 🤨 The difference is that I used local `gradle` installed and I have it in version 8.6. But it shouldn't matter, as the task parameters point to 8.8.

I'll go ahead and cherry-pick your commit and merge this PR.

ParaskP7 commented:
> ...it looks like my update updated jar to version 8.6 🤨

Oh, interesting, and I was wondering what was happening there, thanks for sharing! 😅 

> I'll go ahead and cherry-pick your commit and merge this PR.

🚀
